Source/WebCore/platform/gstreamer/GStreamerElementHarness.cpp

```diff
@@ -659,7 +659,7 @@ String MermaidBuilder::describeCaps(const GRefPtr<GstCaps>& caps)
     for (unsigned i = 0; i < capsSize; i++) {
         auto* features = gst_caps_get_features(caps.get(), i);
         const auto* structure = gst_caps_get_structure(caps.get(), i);
-        builder.append(gstStructureGetName(structure).toString(), "<br/>"_s);
+        builder.append(toString(gstStructureGetName(structure)), "<br/>"_s);
```

Comment:
This is an example where we do not want to create a string just so we can append. The append function already takes `span<const char8_t>` so it's a waste to allocate and destroy a temporary string:
```suggestion
        builder.append(gstStructureGetName(structure).span(), "<br/>"_s);
```

Reply:
My question is, that I could not check for myself, what happens if we have UTF8 characters when appending? Will they be properly converted? Or do you mean that we don't want to explicitly create a String but it will implicitly created?

Reply:
> My question is, that I could not check for myself, what happens if we have UTF8 characters when appending? Will they be properly converted?

Yes, they will be properly converted, and without creating any additional unnecessary copies, as part of the concatenations process used by `makeString` and `StringBuilder::append`.

There are tests that cover this and you are also welcome to add more if you have doubts.

Reply:
Nice to know! I will add a small test just in case.